File: stepik-union/src/main/java/org/stepik/plugin/refactoring/rename/StepikRenameDialog.java
package org.stepik.plugin.refactoring.rename;

import com.intellij.openapi.editor.Editor;
import com.intellij.openapi.project.Project;
import com.intellij.psi.PsiElement;
import com.intellij.refactoring.rename.RenameDialog;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

/**
 * @author meanmail
 */
public class StepikRenameDialog extends RenameDialog {

    private static final String MESSAGE = "Don't make any changes to the structure of the project tree, it might become defective.";

    public StepikRenameDialog(
            @NotNull Project project,
            @NotNull PsiElement psiElement,
            @Nullable PsiElement nameSuggestionContext, Editor editor) {
        super(project, psiElement, nameSuggestionContext, editor);
    }

    @NotNull
    @Override
    protected String getLabelText() {
        return MESSAGE;
    }

    @Override
    protected boolean hasPreviewButton() {
        return false;
    }

    @Override
    protected boolean areButtonsValid() {
        return false;
    }
}

File: stepik-union/src/main/java/org/stepik/plugin/refactoring/rename/StepikRenameDialog.kt
package org.stepik.plugin.refactoring.rename

import com.intellij.openapi.editor.Editor
import com.intellij.openapi.project.Project
import com.intellij.psi.PsiElement
import com.intellij.refactoring.rename.RenameDialog


internal class StepikRenameDialog(
        project: Project,
        psiElement: PsiElement,
        nameSuggestionContext: PsiElement?,
        editor: Editor) : RenameDialog(project, psiElement, nameSuggestionContext, editor) {

    override fun getLabelText(): String {
        return MESSAGE
    }

    override fun hasPreviewButton(): Boolean {
        return false
    }

    override fun areButtonsValid(): Boolean {
        return false
    }

    companion object {

        private val MESSAGE = "Don't make any changes to the structure of the project tree, it might become defective."
    }
}
